apps/public-docsite-v9/src/Concepts/Slots/Slots.stories.mdx

```diff
@@ -0,0 +1,350 @@
+
+When a slot is defined in the props of a component, it can be set to null to prevent the slot from being rendered.
+In cases where components need to require a slot to always be rendered,
+it can be wrapped in `NonNullable<T>`.
+
+For example, in `RadioButton` the indicator must always be rendered.
+
+```ts
+indicator: NonNullable<Slot<'div'>>;
+```
+
+Non-nullable and optional/required slots are independent from one another.
+A slot can be defined with as optional or required and nullable or non-nullable.
+
+#### The special root slot
+
+Every component has a `root` slot.
+You can think of the root slot as
+the element the component renders that contains all component's rendered content.
+
+The root slot
+
+- defines the type of element that will be rendered
+- always gets `className` and `style` properties
+- get the properties passed to the component
+
+### Styling and rendering components with slots
+
+You can think of each slot as container for props and style that will be rendered as an element.
+
+The Fluent UI hooks architecture breaks up component rendering primarily into 3 parts
+
+- use*Component*(): This takes in props and produces state
+- use*Component*Styles() - This uses state to define and apply class styles
+- render*Component*() - This renders the elements of the component
+
+> If you see the suffix `_unstable` that means that the API may have a breaking change in the future.
+> It does **not** mean the code is unstable or unfit for production.
+
+`Button` follows this pattern:
+
+- `useButton` will take in `ButtonProps` and return `ButtonState`.
+
+```ts
+const useButton_unstable = (
+  props: ButtonProps,
+  ref: React.Ref<HTMLButtonElement | HTMLAnchorElement>,
+): ButtonState
+```
+
+Hooks like `useButton` calculate state from props. Typically, this involves filling in default values, adding event handlers,
+and define the default element types for each slot.
+
+- `useButtonStyles` takes in and mutates `ButtonState` adding classNames to each slot.
+
+```ts
+const useButtonStyles_unstable = (state: ButtonState)
+```
+
+Hooks like `useButtonStyles` create classes and conditionally apply them based on the input state. Classes are typically created and applied with `makeStyles` and `mergeClasses`.
+For instance, if a component is disabled then disabled styles are added.
+
+- `renderButton` takes in `ButtonState` and renders content.
+
+```tsx
+const renderButton_unstable = (state: ButtonState) => {
+  const { slots, slotProps } = getSlots<ButtonSlots>(state);
+  const { iconOnly, iconPosition } = state;
+
+  return (
+    <slots.root {...slotProps.root}>
+      {iconPosition !== 'after' && slots.icon && <slots.icon {...slotProps.icon} />}
+      {!iconOnly && state.root.children}
+      {iconPosition === 'after' && slots.icon && <slots.icon {...slotProps.icon} />}
+    </slots.root>
+  );
+};
+```
+
+The `getSlots` method splits up the state to return the slot elements to render and the props to apply.
```

Comment:
```suggestion
The `getSlots` method splits up the state to return the slot elements to render and the props to apply to each slot.
```

small suggestion that was slightly clearer in my brain -- up to you whether you think it's better :)